apps/frontend/src/app/2_molecules/BalanceInput/BalanceInput.tsx

```diff
@@ -0,0 +1,44 @@
+import React from 'react';
+import { forwardRef, useImperativeHandle, useRef } from 'react';
+
+import { BigNumberish } from 'ethers';
+
+import { SupportedTokens } from '@sovryn/contracts';
+import { ChainId } from '@sovryn/ethers-provider';
+import { AmountInput, AmountInputProps } from '@sovryn/ui';
+
+import { useMaxAssetBalance } from '../../../hooks/useMaxAssetBalance';
+import { getRskChainId } from '../../../utils/chain';
+import { fromWei } from '../../../utils/math';
+
+const DEFAULT_GAS_LIMIT = 1000000; // 1M wei
+
+type BalanceInputProps = Omit<AmountInputProps, 'maxAmount' | 'unit'> & {
+  token: SupportedTokens;
+  gasLimit?: BigNumberish;
+  chainId?: ChainId;
+};
+
+export const BalanceInput = forwardRef<HTMLInputElement, BalanceInputProps>(
```

Comment:
I don't see a reason for having this component as it just wraps `AmountInput` and sets `maxAmount` prop but in majority of the cases, you will need to have that `maxAmount` value in the parent component and use it for a label/max button as well.

Reply:
You are right, without "MAX" button it not useful at all. Will remove it.